realm/realm-library/src/main/java/io/realm/internal/TableQuery.java

```diff
@@ -16,29 +16,33 @@
 
 package io.realm.internal;
 
+import android.util.Log;
+
 import java.util.Date;
 
 import io.realm.Case;
 import io.realm.Sort;
 
 
 public class TableQuery implements NativeObject {
-    protected boolean DEBUG = false;
+    private static final String TAG = "TABQ";
+    private static final boolean DEBUG = false;
 
-    protected long nativePtr;
     private static final long nativeFinalizerPtr = nativeGetFinalizerPtr();
-    protected final Table table;
+    private final Table table;
     private final Context context;
+    private final long nativePtr;
 
     // All actions (find(), findAll(), sum(), etc.) must call validateQuery() before performing
     // the actual action. The other methods must set queryValidated to false in order to enforce
     // the first action to validate the syntax of the query.
     private boolean queryValidated = true;
 
     // TODO: Can we protect this?
+    // FIXME: Either explain why we need the context, of quit holding on to it.
```

Comment:
It is described in the comments. See https://github.com/realm/realm-java/blob/master/realm/realm-library/src/main/java/io/realm/internal/Context.java#L22
So i think the FIXME is not needed here?